hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java

```diff
@@ -231,36 +233,41 @@ private void decorateConf() {
     }
   }
 
+  @Override public String toString() {
+    return "ReplicationSource[peerId=" + peerId + "]";
+  }
+
   @Override
   public void enqueueLog(Path wal) {
     if (!this.filterInWALs.test(wal)) {
       LOG.trace("NOT replicating {}", wal);
       return;
     }
-    String logPrefix = AbstractFSWALProvider.getWALPrefixFromWALName(wal.getName());
-    PriorityBlockingQueue<Path> queue = queues.get(logPrefix);
+    // Use WAL prefix as the WALGroupId for this peer.
+    String walGroupId = AbstractFSWALProvider.getWALPrefixFromWALName(wal.getName());
```

Comment:
Better still name it walPrefix? The method is called getWALPrefixFromWALName, name it walGroupId adds new constraint.

Reply:
Yes. Better. Done in next patch.